<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="model" default="$(find car_sim)/urdf/test_robot2_assem.xacro" />
  <param name="robot_description" command="$(find xacro)/xacro $(arg model) --inorder"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <node name="car_control" pkg="car_sim" type="car_control"/>  

  <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen" args="-param robot_description -urdf -model test_robot2_assem"/>

  <rosparam command="load" file="$(find car_sim)/config/diff_drive_controller.yaml" ns="/test_robot2_assem" />
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" ns="/test_robot2_assem" args="diff_drive_controller" />

</launch>
